src/main/java/io/cryostat/jfr/datasource/server/JfrResource.java

```diff
@@ -178,6 +210,36 @@ private void setFile(
         }
     }
 
+    private List<String> deleteAllFiles() {
+        File dir = new File(jfrDir);
+        final List<String> deleteFiles = new ArrayList<>();
+        if (dir.exists() && dir.isDirectory()) {
+            for (File f : dir.listFiles()) {
+                if (f.isFile()) {
+                    try {
+                        Files.delete(f.toPath());
+                        deleteFiles.add(f.getName());
+                        LOGGER.info("Deleted: " + f.getName());
+                    } catch (IOException e) {
```

Comment:
These exceptions should at least be logged, even if they are not rethrown. But I would also consider that perhaps they should be rethrown. If the exception occurs because the file does not exist then this could result in a `404` like I said above. Otherwise, if some other I/O failure occurs, then that's probably an `HTTP 500`.

Reply:
Sure I will add some handling to these.